src/fft.rg

```diff
@@ -144,133 +152,435 @@ function fft.generate_fft_interface(itype, dtype)
+    var output_base = get_base_out(rect_t_out(output.ispace.bounds), __physical(output)[0], __fields(output)[0])
+    var offset_in = get_offset_in(rect_t_in(input.ispace.bounds), __physical(input)[0], __fields(input)[0])
+
+    --Set idist: idist is the distance between the first element of two consecutive batches
+    --In a transform where each batch is a 256x256 complex64 transform, offset_1 will be 16, offset_2 will be 16*256, and offet_3 willbe 16*256*256
+    --idist should be 256*256 in this case, so we want offset_3/offset_1
+    var offset_1 = offset_in[0].offset
+    var offset_2 = offset_in[1].offset
+    var offset_3 = offset_in[2].offset
+    var i_dist = offset_3/offset_1
+
+    format.println("Offset 1 = {}, Offset 2 = {}, Offset 3 = {}", offset_1, offset_2, offset_3)
+
     var lo = input.ispace.bounds.lo:to_point()
     var hi = input.ispace.bounds.hi:to_point()
-    var flags = fftw_c.FFTW_MEASURE
-    p.p = plan_dft(
-      [data.range(dim):map(function(i) return rexpr hi.x[i] - lo.x[i] + 1 end end)],
-      [&fftw_c.fftw_complex](input_base),
-      [&fftw_c.fftw_complex](output_base),
-      fftw_c.FFTW_FORWARD,
-      flags)
+
+    --dtype size is 8 for complex32 and 16 for complex64
+    format.println("Size of dtype is {}", dtype_size)
+
+    --If GPUs, call make_plan_gpu_batch
+    if gpu_available then
+      format.println("Num_local_gpus is {}", iface.get_num_local_gpus())
+      if iface.get_num_local_gpus() > 0 then
+        format.println("GPUs identified: calling make_plan_gpu_batch...")
+        make_plan_gpu_batch(input, output, plan, p.address_space)
+      end 
+    end
+
+    --Call fftw_plan fftw_plan_many_dft(int rank, const int *n, int howmany, fftw_complex *in, const int *inembed, int istride, int idist, fftw_complex *out, const int *onembed, int ostride, int odist, int sign, unsigned flags)
+    --Taking a transform where {256x256x7} is passed (i.e. 7 batches of 256x256), the parameters should be as follows:
+      --rank[In] – How many dimensions a single transform has: 2
+      --n[In] - Array of size rank, describing the size of each dimension: {256, 256}
+      --inembed[In] – Array of size rank. Here we pass the real dimensions of the input array. It should contain the dimension size + padding: {256, 256} again, because input is not padded.
+      --istride[In] – Stride between two consecutive elements in lowest dimension: 1
+      --idist[In] – Distance between two input batches: 256 * 256
+      --onembed[In] – Array of size rank. Here we pass the real dimensions of the output array. It should contain the dimension size + padding: {256, 256} again, because input is not padded.
+      --ostride[In] – Stride between two consecutive elements in lowest dimension: 1
+      --odist[In] – Distance between two output batches: 256 * 256
+      --type[In] – The transform data type: Z2Z or D2Z.
+      --batch[In] – How many batches should be computed: 7
```

Comment:
Is this accurate? I don't think this matches the FFTW interface.

Reply:
You're right -- this was copied from CUFFT, although i think it's fairly similar. I will remove this.